test/data/shared-job-annotations.yaml

```diff
@@ -0,0 +1,13 @@
+shared:
+    annotations:
+        foo: a
+        bar: b
+
+jobs:
+    main:
+        annotations:
+            foo: c
+            bar: d
```

Comment:
Can we modify this so that the job level has `foo`, and not `bar`? So that we know it will pick up `bar` from `shared`, and not completely override.